### ISSUE TITLE
Support correlation-context in absence of Request-Id or traceparent

### ISSUE THREAD

**Issue by lmolkova** [microsoft/ApplicationInsights-aspnetcore#901]
Correlation-Context is used by partner teams in Microsoft to propagate arbitrary key-value pairs along with request flow and in addition to tracing identifiers (Request-Id, cV or traceparent).

When 1P app instrumented with cV attempts to work with app that is instrumented with ApplicationInsights, AppInsights ignores Correlation-Context.
It happens because Correlation-Context according to spec is an extension on top of Request-Id (or traceparent) and is not read when Request-Id is not present.

We should read Correlation-Context regardless of Request-Id (traceparent) presence.

**Comment by cijothomas** [comment]
@lmolkova Do we need this for upcoming release?

**Comment by lmolkova** [comment]
yep, we have a partner team asking us for it (just got confirmation today). Do you want to do it along with w3c changes? 

If not, that's fine - I wanted to volunteer.
